gate-web/src/test/groovy/com/netflix/spinnaker/gate/config/GateCorsAllowedOriginConfigSpec.groovy

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.config
+
+import com.netflix.spinnaker.gate.Main
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.TestPropertySource
+import org.springframework.test.web.servlet.MockMvc
+import spock.lang.Specification
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.header
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+
+@AutoConfigureMockMvc
+@SpringBootTest(classes = Main)
+@ActiveProfiles('alloworigincors')
+@TestPropertySource(properties = ["spring.config.location=classpath:gate-test.yml"])
+class GateCorsAllowedOriginConfigSpec extends Specification {
+
+  @Autowired
+  private MockMvc mvc
+
+  def "cors filter will send 403 back for localhost"() {
+    expect: "Status is 403 and the VARY header comes in as expected"
+    mvc.perform(get("/health").header('Origin', 'https://localhost'))
+      .andExpect(status().is(403))
```

Comment:
in the regex filter implementation, we return a 200 OK with `Access-Control-Allow-Origin: *`, which makes the browser not propagate the cookie as a result (and eventually the request fails as a result).

in this case if we explicitly return a 403 Forbidden I suspect it has the same eventual outcome, but we should run through this interaction from the browser side and see how it behaves

Reply:
When we send 200 OK back we served the first req coming in from a different origin but the subsequent requests failed because cookie won't be transmitted ?  Correct ?

Reply:
that is correct - I think we send a 200 OK on the OPTIONS preflight from the browser, but when it comes back with no cookie it would trigger an auth redirect most likely